### ISSUE TITLE
Improve chamfer loss function

### ISSUE THREAD

**Issue by LucasColas** [facebookresearch/pytorch3d#2017]
## 🚀 Feature
Hello everyone,
My feature is to actually improve an existing one. I took a look at the implementation of the chamfer loss function. 
Wouldn't it be to great to fully implement it on GPU ?

## Motivation

 It would be great if one could chamfer loss function implemented on GPU. It can be easily parallelized on GPU. And it'd be faster to compute. But I'm note sure if it'd be faster than the current implement in pytorch3D.

## Pitch
It could be implemented based on this code : https://github.com/ark1234/NeurIPS2023-InfoCD/blob/main/codes/Chamfer3D/chamfer3D.cu
